examples/remote/binary_dependencies/cargo/crates.bzl

```diff
@@ -9,6 +9,144 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")  # bui
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")  # buildifier: disable=load
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")  # buildifier: disable=load
 
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of normal dependencies for the Rust targets of that package.
+_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+        "ferris-says": "@remote_binary_dependencies__ferris_says__0_2_0//:ferris_says",
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of proc_macro dependencies for the Rust targets of that package.
+_PROC_MACRO_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of normal dev dependencies for the Rust targets of that package.
+_DEV_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+# EXPERIMENTAL -- MAY CHANGE AT ANY TIME: A mapping of package names to a set of proc_macro dev dependencies for the Rust targets of that package.
+_DEV_PROC_MACRO_DEPENDENCIES = {
+    "remote/binary_dependencies": {
+    },
+}
+
+def crates(deps):
```

Comment:
How is this intended to be used? I don't see any usages in this pr.

Reply:
I posted this on another comment but:
```python
load("//third_party/cargo:crates.bzl", "crates")
rust_library(
    name = "example1",
    deps = [
        "//third_party/cargo:dependency1",
    ] + crates("dependency2"),,
    #...
)

rust_library(
    name = "example2",
    deps = crates([
        "dependency1"
        "dependency2"
    ]),
    #...
)
```

`example1` and `example2` are the same crate, just with different names for the sake of the example

Reply:
Even with this example, the semantics here are not clear. 

Pseudocode of something I'd find clearer

```
resolve_crate_dependences("crate-name", [
  "serde"
])
```

If you insist upon the magical version, that can be derived from the non-magical version

```
resolve_crate_dependencies_here([
  "serde"
])
```

where

```
def resolve_crate_dependencies_here(crate_names):
  return resolve_crate_dependencies(native.package_name(), crate_names)
```

Reply:
Marking resolved per other discussion